nuclio/config.py

```diff
@@ -203,15 +203,49 @@ def set_env(config, env):
             raise ValueError(
                 'cannot parse environment value from: {}'.format(line))
 
+        # TODO: change to allow magic secrets
         update_env_var(config, key, value)
 
 
 def set_env_dict(config, env={}):
     for k, v in env.items():
-        update_env_var(config, k, str(v))
+        update_env_var(config, k, value=str(v))
+
+
+def set_secrets_dict(config, secrets={}):
+    for k, v in secrets.items():
+        try:
+            v_dict = json.loads(v)
+        except ValueError as e:
+            logger.warning(f'failed to deserialize env variable from secret: {v}')
+            raise e
+
+        secret_key_ref = v_dict.get('secretKeyRef', {}) or v_dict.get('secret_key_ref', {})
```

Comment:
:eyes: in what case would the input be camelCase and what snake case?

Reply:
for now we only need CamelCase so changed to only that. might need when supporting magic env vars to use snake_case, will change and add comment if that's the case.